ingestion/functions/common/parsing_lib.py

```diff
@@ -25,6 +25,9 @@
 AUTH_FIELD = "auth"
 
 MAX_WAIT_TIME = 600  # 10 minutes maximum wait for response
+MAX_CONN_TRIES = 3  # Number of connection attempts to data service
+CONN_WAIT = 600  # Wait time between data service connection attempts
```

Comment:
is this a bit pessimistic? The situation is "the data service appears to sometimes not be responding so we'll try a few times", is ten minutes too long to wait? How about exponential backoff from a couple of seconds up to a couple of minutes? Or do we know it takes 10 minutes for the data service to restart itself/start accepting connections again?

Reply:
Added another exponential backoff counting up to 15 minutes